frontend/src/test/java/com/google/tripmeout/frontend/servlet/PlaceVisitParentServletTest.java

```diff
@@ -0,0 +1,195 @@
+                                                .setUserMark(PlaceVisitModel.UserMark.YES)
+                                                .build();
+
+  private static final PlaceVisitModel UK_B =
+      PlaceVisitModel.builder()
+          .setId("123")
+          .setPlaceName("UK Trip")
+          .setTripId("123abc")
+          .setPlacesApiPlaceId("ChIJ3S-JXmauEmsRUcIaWtf4MzE")
+          .setUserMark(PlaceVisitModel.UserMark.MAYBE)
+          .build();
+
+  @Before
+  public void setUp() {
+    initMocks(this);
+    Injector injector = Guice.createInjector(new GsonModelSerializationModule());
+    gson = injector.getInstance(Gson.class);
+  }
+
+  @Test
+  public void doPost_badUri_setsInternalServerError() throws IOException {
+    FakeHttpServletResponse response = new FakeHttpServletResponse();
+    PlaceVisitStorage placeStorage = new InMemoryPlaceVisitStorage();
+    PlaceVisitParentServlet servlet = new PlaceVisitParentServlet(placeStorage, gson, placeService);
+    when(request.getRequestURI()).thenReturn("/trips-number/abc123/placeVisits");
+    when(request.getReader()).thenReturn(new BufferedReader(new StringReader(gson.toJson(UK))));
+
+    servlet.doPost(request, response);
+
+    assertThat(response.getStatus()).isEqualTo(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+  }
+
+  @Test
+  public void doPost_badPlaceId_setsNotFoundStatus()
+      throws IOException, InvalidPlaceIdException, PlacesApiRequestException {
+    FakeHttpServletResponse response = new FakeHttpServletResponse();
+    PlaceVisitStorage placeStorage = new InMemoryPlaceVisitStorage();
+    PlaceVisitParentServlet servlet = new PlaceVisitParentServlet(placeStorage, gson, placeService);
+
+    doThrow(InvalidPlaceIdException.class).when(placeService).validatePlaceId("NY, USA");
+    when(request.getRequestURI()).thenReturn("/trips/abc123/placeVisits");
+    when(request.getReader()).thenReturn(new BufferedReader(new StringReader(gson.toJson(NY))));
+
+    servlet.doPost(request, response);
+
+    assertThat(response.getStatus()).isEqualTo(HttpServletResponse.SC_BAD_REQUEST);
+  }
+
+  @Test
+  public void doPost_goodRequest_returnsAddedPlaceVisitModel()
+      throws IOException, ApiException, InterruptedException {
+    FakeHttpServletResponse response = new FakeHttpServletResponse();
+    PlaceVisitStorage placeStorage = new InMemoryPlaceVisitStorage();
+    PlaceVisitParentServlet servlet = new PlaceVisitParentServlet(placeStorage, gson, placeService);
+    when(request.getRequestURI()).thenReturn("/trips/abc123/placeVisits");
+    when(request.getReader()).thenReturn(new BufferedReader(new StringReader(gson.toJson(NY))));
+
+    servlet.doPost(request, response);
+    PlaceVisitModel place = gson.fromJson(response.getResponseString(), PlaceVisitModel.class);
+
+    assertThat(response.getStatus()).isEqualTo(HttpServletResponse.SC_CREATED);
+    assertThat(place.placeName()).isEqualTo("NY Trip");
```

Comment:
Nit: in the future, please use a single assert call on an instantiated model object that has the expected results (instead of manually checking individual fields).